Bottom: 1208a8b66a2611a6f4f7512652ae17c5974ca83d
Top:    f8a160520ed41187ee12626a9f821a8038a2c3b8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-03 22:02:07 +0000

Fix z-coord when flipping the board


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index feed8b7..5065ac0 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1800,7 +1800,8 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
 
   glPushMatrix ();
   glScalef ((ghid_flip_x ? -1. : 1.) / gport->zoom,
-            (ghid_flip_y ? -1. : 1.) / gport->zoom, 1);
+            (ghid_flip_y ? -1. : 1.) / gport->zoom,
+            (ghid_flip_x == ghid_flip_y) ? 1. : -1.);
   glTranslatef (ghid_flip_x ? gport->view_x0 - PCB->MaxWidth  :
                              -gport->view_x0,
                 ghid_flip_y ? gport->view_y0 - PCB->MaxHeight :
